starter-code/scripts/models/article.js

```diff
@@ -69,18 +69,31 @@ Article.getAll = function(nextFunction) {
 Article.numWordsAll = function() {
   return Article.allArticles.map(function(currentArticle) {
     return currentArticle.body.match(/\w+/g).length;
-  }).reduce(function() {
-    // TODO: complete this function to sum up all of the words.
-  })
+  }).reduce(function(acc, current) {
+    // TODO: DONE complete this function to sum up all of the words.
+    return acc + current
+  }, 0)
 }
 
 Article.allAuthors = function() {
-  // TODO: return a mapped collection
-      // with just the author names
-
-      //then chain reduce, and set the accumulator to an array
-      // to build a unique list of author names.
+  return Article.allArticles.map(function(allArticles) {
+    return allArticles.author
```

Comment:
Great use of map to return an array of authors (strings)! You are missing a semicolon here, but it's potentially because there was a bad .eslintrc file in the root of this lab, so I won't mark you off for that.